### PR TITLE
Liquidity provider integration

### PR DESCRIPTION
**Pull request by PirosB3**
This PR upgrades 0x API to the latest `AssetSwapper` (package is not released yet) and provides a path to add private liquidity providers to 0x API. Private liquidity providers can now be added to 0x API by setting the `LIQUIDITY_POOL_REGISTRY_ADDRESS` environment variable to the liquidity provider's registry contract.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -114,6 +124,7 @@ const sourceFees: { [key in ERC20BridgeSource]: BigNumber } = {
     [ERC20BridgeSource.CurveUsdcDaiUsdtTusd]: new BigNumber(8e5),
     [ERC20BridgeSource.CurveUsdcDaiUsdtBusd]: new BigNumber(8e5),
     [ERC20BridgeSource.Kyber]: new BigNumber(8e5),
+    [ERC20BridgeSource.LiquidityProvider]: new BigNumber(8e5),
```
Comment (PirosB3) [inline src/config.ts]: @fragosti I'm really unsure what I should be putting here as a source fee, who should I be asking?

Comment (dekz) [reply src/config.ts]: How much gas does a LiquidityProvider trade cost? Then add an additional 150e3

Comment (PirosB3) [reply src/config.ts]: @hysz much gas does a LiquidityProvider trade cost?

File: package.json [hunk 2]
```diff
@@ -27,7 +27,9 @@
         "lint": "tslint --project . --format stylish"
     },
     "resolutions": {
-        "@0x/contract-addresses": "0xProject/gitpkg-registry#0x-contract-addresses-v4.9.0-b8ec7f5e2"
+        "@0x/contract-addresses": "0xProject/gitpkg-registry#0x-contract-addresses-v4.9.0-58b6c25d5",
+        "@0x/contract-wrappers": "0xProject/gitpkg-registry#0x-contract-wrappers-v13.6.3-58b6c25d5",
+        "@0x/asset-swapper": "0xProject/gitpkg-registry#0x-asset-swapper-v4.4.0-58b6c25d5"
```
[the remark this answers is not in the document]
Comment (PirosB3) [reply package.json]: fixed

File: src/config.ts (hunk 1 above)
Comment (hysz) [reply src/config.ts]: The total gas cost for a LiquidityProvider trade routed through the 0x Exchange contract should cost ~250k gas. Note that as new liquidity providers come online we may have to adjust this value.

File: src/config.ts [hunk 3]
```diff
@@ -89,6 +89,16 @@ export const LOGGER_INCLUDE_TIMESTAMP = _.isEmpty(process.env.LOGGER_INCLUDE_TIM
     ? DEFAULT_LOGGER_INCLUDE_TIMESTAMP
     : assertEnvVarType('LOGGER_INCLUDE_TIMESTAMP', process.env.LOGGER_INCLUDE_TIMESTAMP, EnvVarType.Boolean);
 
+export const LIQUIDITY_POOL_REGISTRY_ADDRESS: string | undefined = _.isEmpty(
+    process.env.LIQUIDITY_POOL_REGISTRY_ADDRESS,
+)
+    ? undefined
+    : assertEnvVarType(
+          'LIQUIDITY_POOL_REGISTRY_ADDRESS',
```
[the remark this answers is not in the document]
Comment (fragosti) [reply src/config.ts]: Sounds like we're trying to keep this in the DL.

[the remark this answers is not in the document]
Comment (PirosB3) [reply src/config.ts]: what is DL?

**Comment by PirosB3** [comment]
@fragosti thank you for fixing CI

File: src/config.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (hysz) [reply src/config.ts]: @PirosB3 since liquidity providers will likely use their own registry it could be good to make this an array.

**Pull request merged**